src/Engine/Component/Component.cpp

```diff
@@ -10,12 +9,11 @@
 #include <Engine/Renderer/RenderObject/RenderObjectManager.hpp>
 #include <Engine/System/System.hpp>
 
-namespace Ra {
-namespace Engine {
+namespace Ra::Engine {
 
 using namespace Core::Utils; // log
 
-Component::Component( const std::string& name, Entity* entity ) : m_name{name}, m_entity{entity} {
+Component::Component( std::string name, Entity* entity ) : m_name(std::move(name)), m_entity{entity} {
```

Comment:
This logic (no reference + move) to pass std::string in constructors must be added in the code style documentation.
Codacy is not happy about it btw https://app.codacy.com/app/STORM/Radium-Engine/file/36961256903/issues/source?fileBranchId=13785087#l28